demos/file/README.md

```diff
@@ -0,0 +1,59 @@
+# File
+
+In this demo we manipulate the data on the host's file system with syscalls. 
+
+Since AMT does not have a file system, we use semihosting to implement the operations for files.
+
+To describe the concepts, we need to introduce:
+
+* `open` (in `src/user/file.c`)
+* `close` (in `src/user/file.c`)
+* `read` (in `src/user/file.c`)
+* `write` (in `src/user/file.c`)
+* `remove` (in `src/user/file.c`)
+
+## `open`
+
+This function opens a file descriptor according to the `path` and the `oflags` we specified.
+
+In AMT, the permissions we provide are `O_RDONLY` and `O_WRONLY`.
+
+If we can open a file, the return value will be nonzero. If we can not open a file, the value will be set to `-1`.
+
+## `close`
+
+We can close a file descriptor by using this function. 
+
+If the operation succeeds, the return value will be `0`. If the operation failed, the return value will be `-1`
+
+## `read`
+
+`read()` attempts to read at most `nbyte` bytes from file descriptor `fildes` into the buffer starting at `buf`.
+
+On success, the number of bytes read is returned, and the file position is advanced by this number.
+
+## `write`
+
+We can use `write()` to write up to `nbyte` bytes into file descriptor `fildes` from the buffer starting at `buf`.
+
+The return value will be the number of bytes written on success.
```

Comment:
This leaves out the failure value, which is -1 like the rest. (https://man7.org/linux/man-pages/man2/write.2.html)